src/main/java/com/linkedin/kafka/clients/largemessage/ConsumerRecordsProcessor.java

```diff
@@ -58,22 +62,25 @@ public ConsumerRecordsProcessor(MessageAssembler messageAssembler,
    * @param consumerRecords The consumer records to be filtered.
    * @return filtered consumer records.
    */
-  public ConsumerRecords<K, V> process(ConsumerRecords<byte[], byte[]> consumerRecords) {
-    Map<TopicPartition, List<ConsumerRecord<K, V>>> filteredRecords = new HashMap<>();
+  public ConsumerRecordsProcessResult<K, V> process(ConsumerRecords<byte[], byte[]> consumerRecords) {
+    ConsumerRecordsProcessResult<K, V> result = new ConsumerRecordsProcessResult<>();
     for (ConsumerRecord<byte[], byte[]> record : consumerRecords) {
       TopicPartition tp = new TopicPartition(record.topic(), record.partition());
-      ConsumerRecord<K, V> handledRecord = handleConsumerRecord(record);
-      // Only put record into map if it is not null
-      if (handledRecord != null) {
-        List<ConsumerRecord<K, V>> list = filteredRecords.get(tp);
-        if (list == null) {
-          list = new ArrayList<>();
-          filteredRecords.put(tp, list);
+      if (result.hasError(tp)) {
```

Comment:
We don't have to. This is to avoid refetching the messages. So we only stop if there is an exception in the partition.